src/serializer.ts

```diff
@@ -32,7 +36,38 @@ export class Serializer {
      * Gets the discriminator field for a given class.
      */
     private static getParentOptions(clazz: Instantiable): ParentOptions | undefined {
-        return Reflect.getMetadata('serializer:parent', clazz);
+        return Reflect.getMetadata(METADATA_PARENT, clazz);
+    }
+
+    /**
+     * Returns the fields used to map data properties on result's ones.
+     *
+     * @param instance An instance of the class we're using.
+     * @param obj The current object we're deserializing.
+     * @returns A custom array containing obj's field as index and corresponding result's field as value.
+     */
+    private static getPropertyMap(obj: any, instance: any): { [index: string]: string } {
```

Comment:
Making a method static removes it from the instance, so if you have X instance of the class, only one method will be created. And I made it static because of WebStorm too to be honest.